lib/primitives/elementary/straight_line.rb

```diff
@@ -1,13 +1,40 @@
+# frozen_string_literal: true
+
 module Primitives
   module Elementary
     # Ax + By + C = 0
     class StraightLine
-      attr_reader :a
-      attr_reader :b
-      attr_reader :c
+      attr_reader :a, :b, :c
 
       def initialize(a, b, c)
-        # Your code goes gere...
+        check_argument(a)
+        check_argument(b)
+        check_argument(c)
+        @a = a
+        @b = b
+        @c = c
+      end
+
+      # @return distance between this line and point
+      def shortest_distance_point(point)
+        raise TypeError, "Invalid type of argument (Must be Point)" unless point.is_a? Point
+
+        ((a * point.x + b * point.y + c).abs / Math.sqrt((a**2) + (b**2)))
+      end
+
+      # @return point if straight_lines intersect, otherwise nil
+      def intersect_check(straigt_line)
+        raise TypeError, "Invalid type of argument (Must be Point)" unless straigt_line.is_a? StraightLine
+
+        if (@a * straigt_line.b - straigt_line.a * @b) != 0

```

Comment:
This condition may be always true if point\`s coordinates and/or coefficients are floating-point numbers.
```ruby
a = 0.1
b = 1 - 0.9  # 0.09999999999999998 
a == b  # false
```
Reference: [Floating-point arithmetic](https://en.wikipedia.org/wiki/Floating-point_arithmetic).